templates/accounts.go

```diff
@@ -26,67 +26,118 @@ import (
-  prepare(signer: AuthAccount) {
-	let acct = AuthAccount(payer: signer)
+transaction(publicKeys: [[UInt8]]) {
+	prepare(signer: AuthAccount) {
+		let acct = AuthAccount(payer: signer)
 
-	for key in publicKeys {
-	  acct.addPublicKey(key)
+		for key in publicKeys {
+			acct.addPublicKey(key)
+		}
 	}
-	
-	if code.length > 0 {
-	  acct.setCode(code)
+}
+`
+
+const createAccountWitContractsTemplate = `
+transaction(publicKeys: [[UInt8]], contracts: {String: [UInt8]}) {
+	prepare(signer: AuthAccount) {
+		let acct = AuthAccount(payer: signer)
+
+		for key in publicKeys {
+			acct.addPublicKey(key)
+		}
+
+		for contract in contracts.keys {
+			acct.contracts.add(name: contract, code: contracts[contract]!)
+		}
 	}
-  }
 }
 `
 
 // CreateAccount generates a transactions that creates a new account.
 //
-// This template accepts a list of public keys and a code argument, both of which are optional.
+// This template accepts a list of public keys and a contracts argument, both of which are optional.
+//
+// The contracts argument is a dictionary of *contract name*: *contract code (in bytes)*.
+// All of the contracts will be deployed to the account.
 //
 // The final argument is the address of the account that will pay the account creation fee.
 // This account is added as a transaction authorizer and therefore must sign the resulting transaction.
-func CreateAccount(accountKeys []*flow.AccountKey, code []byte, payer flow.Address) *flow.Transaction {
+func CreateAccount(accountKeys []*flow.AccountKey, contracts []Contract, payer flow.Address) *flow.Transaction {
 	publicKeys := make([]cadence.Value, len(accountKeys))
 
 	for i, accountKey := range accountKeys {
 		publicKeys[i] = bytesToCadenceArray(accountKey.Encode())
 	}
 
+	contractKeyPairs := make([]cadence.KeyValuePair, len(contracts))
+
+	for i, contract := range contracts {
+		contractKeyPairs[i] = cadence.KeyValuePair{
+			Key:   cadence.NewString(contract.Name),
+			Value: bytesToCadenceArray(contract.Source),
+		}
+	}
+
 	cadencePublicKeys := cadence.NewArray(publicKeys)
-	cadenceCode := bytesToCadenceArray(code)
+	cadenceContracts := cadence.NewDictionary(contractKeyPairs)
 
 	return flow.NewTransaction().
-		SetScript([]byte(createAccountTemplate)).
+		SetScript([]byte(createAccountWitContractsTemplate)).
 		AddAuthorizer(payer).
 		AddRawArgument(jsoncdc.MustEncode(cadencePublicKeys)).
-		AddRawArgument(jsoncdc.MustEncode(cadenceCode))
+		AddRawArgument(jsoncdc.MustEncode(cadenceContracts))
+}
+
+type Contract struct {
+	Name string
+	Source []byte
+}
+
+const updateAccountContractTemplate = `
+transaction(name: String, code: [UInt8]) {
+	prepare(signer: AuthAccount) {
+		signer.contracts.update__experimental(name: name, code: code)
+	}
 }
+`
 
-const updateAccountCodeTemplate = `
-transaction(code: [UInt8]) {
-  prepare(signer: AuthAccount) {
-	signer.setCode(code)
-  }
+// UpdateAccountContract generates a transaction that updates the contract deployed at an account.
```

Comment:
```suggestion
// UpdateAccountContract generates a transaction that updates a contract deployed at an account.
```

Current phrasing implies only one contract per account.